src/useFormStates.ts

```diff
@@ -42,10 +42,19 @@ type UseFormStatesOpts<T, I> = {
    * Does not need to be stable/useMemo'd.
    */
   map?: (input: Exclude<I, null | undefined>) => T;
+
+  /**
+   * Sets all `ObjectState`s to readOnly.
+   */
+  readOnly?: boolean;
 };
 
-export function useFormStates<T, I = T>(opts: UseFormStatesOpts<T, I>): { getFormState: (input: I) => ObjectState<T> } {
-  const { config, autoSave, getId, map, addRules } = opts;
+type UseFormStatesHook<T, I> = {
+  getFormState: (input: I, opts?: { readOnly?: boolean }) => ObjectState<T>;
```

Comment:
And this would control it per row, like if there were per-row rules about read-only-ness.